backend/router.js

```diff
@@ -2,16 +2,142 @@ const express = require('express')
+                    to the question: ${message}. Do not ask any type of question within your response. Say something to end this thought and mention going on to the next question. `
+                            :
+                            `
+                    You are a interviewer conducting a behavioral interview. Make sure to talk in the first person as if this was a normal conversation between two people; do not include quotes around your response.  
+                    The user is about give their response to a question that you have 
+                    asked. Please provide a relevant reaction to their answer and say what you liked about their response. Remember to keep this concise. Here is the question that was asked: ${context}. Here is the user's response 
+                    to the question: ${message}. Do not ask any type of question within your response. Say something to end this thought.
+                    Lastly, since this was the last question, you are done conducting the interview.
+                    You must always express your thanks for getting to interview the person and mention that there will be post interview feedback via our post interview review on our platform InterviewMe. 
+                    `
+                }
+            ],
+            model: "llama3-70b-8192",
+        });
+
+
+        const botResponse = response.choices[0]?.message?.content || "I didn't understand that.";
+
+
+        // Use regex (regular expression) to extract follow-up question
+        const followUpQuestionMatch = botResponse.match(/follow-up question, (.*)$/);
+        const followUpQuestion = followUpQuestionMatch ? followUpQuestionMatch[1].trim() : null;
+        res.json({
+            response: botResponse,
+            followUp: followUpQuestion
+        });
+    } catch (error) {
+        console.error("Error getting chat completion:", error);
+        res.status(500).json({ error: "Error getting chat completion" });
+    }
+});
+
+
+
+
+
+
+
+
+// Configure Multer for file uploads
+const storage = multer.diskStorage({
+    destination: (req, file, cb) => {
+        cb(null, 'uploads/');
+    },
+    filename: (req, file, cb) => {
+        cb(null, `${Date.now()}-${file.originalname}`);
+    },
+});
+const upload = multer({ storage });
+
+const groqSTT = new groq({
+    apiKey: process.env.GROQ_API_KEY,
+});
+
+app.post('/transcribe', upload.single('file'), async (req, res) => {
+    try {
+        const file = req.file;
+        const translation = await groqSTT.audio.transcriptions.create({
+            file: fs.createReadStream(file.path),
+            model: 'whisper-large-v3',
+            prompt: 'Specify context or spelling',
+            response_format: 'json',
+            temperature: 0.0,
+        });
+
+        // Delete the file after processing if needed
```

Comment:
What's the condition we don't want to delete the file?

Reply:
The file should always be deleted. That was an old comment that I missed. The audio file is sent to the STT api and then is immediately deleted.